### ISSUE TITLE
validation_error_definition schema is incorrect when detecting an error in an array

### ISSUE THREAD

**Issue by silversurfer34** [tiangolo/fastapi#3790]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional, List

from fastapi import FastAPI
from pydantic import BaseModel

class SubItem(BaseModel):
    price: float

class Item(BaseModel):
    price_list: Optional[List[SubItem]] = None


app = FastAPI()


@app.post("/items/")
async def create_item(item: Item):
    return item
```

### Description

Run the sample program and call the /items/ api with this body:
```
{
  "price_list": [
    {
      "price": "zero"
    }
  ]
}
```
You will get a 422 like this 

```
{
  "detail": [
    {
      "loc": [
        "body",
        "price_list",
        0,
        "price"
      ],
      "msg": "value is not a valid float",
      "type": "type_error.float"
    }
  ]
}
```

But this 422 response does not follow the validation_error_definition schema :
```
validation_error_definition = {
    "title": "ValidationError",
    "type": "object",
    "properties": {
        "loc": {"title": "Location", "type": "array", "items": {"type": "string"}},
        "msg": {"title": "Message", "type": "string"},
        "type": {"title": "Error Type", "type": "string"},
    },
    "required": ["loc", "msg", "type"],
}
```

So the 422 response from fastapi does not respect the schema of the response... 

The issue comes from the error in the array, pydantic returns 0 as index of the error, but it is not a string, it is an integer. And it makes sense to return an integer, otherwise we wouldn't know if it is a key in a dict or an index.

As a developer I can workaround this issue by redefining validation_error_definition in my own code

```
from fastapi.openapi.utils import validation_error_definition

validation_error_definition["properties"] = {
    "loc": {
        "title": "Location", "type": "array", "items": {
            "oneOf": [
                {"type": "string"},
                {"type": "integer"}
            ]
        }
    },
    "msg": {"title": "Message", "type": "string"},
    "type": {"title": "Error Type", "type": "string"},
}
```
But it would be better to do the change directly in the code. What is your opinion ?

### Operating System

Windows

### FastAPI Version

0.68.1

### Python Version

3.7.9

**Comment by dconatha** [comment]
I just stumbled across this error by using https://schemathesis.readthedocs.io/en/stable/ and I agree this should be fixed.

The schema should match the schema of Pydantic's "loc" found here:

https://github.com/samuelcolvin/pydantic/blob/5ccbdcb5904f35834300b01432a665c75dc02296/pydantic/error_wrappers.py#L12

**Comment by dconathan** [comment]
I'm working on a PR for this but it breaks lots of tests because the `ValidationError` schema is hardcoded in lots of tests...

**Comment by AbhijithGanesh** [comment]
It passes the tests, so shouldn't this issue be closed?

**Comment by Kludex** [comment]
No. Issues are closed when problems are solved on the last release of the package.

**Comment by AbhijithGanesh** [comment]
Thanks for the clarification

**Comment by tiangolo** [comment]
Thanks for the report @silversurfer34! :nerd_face: 

This was solved on https://github.com/tiangolo/fastapi/pull/3810 by @dconatha :bow: :cake: 

It will be available in FastAPI `0.75.2` released in a couple of hours. :rocket:
